pkg/operator/controller/dns/controller.go

```diff
@@ -718,14 +718,28 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 			return &dns.FakeProvider{}, nil
 		}
 
+		// Read custom service endpoints from the infra status and set when configured
+		// When none configured, the provider will use the default values
+		var serviceEndpointOverrides ibm.ServiceEndpointOverrides
+		for _, endpointConfig := range infraStatus.PlatformStatus.IBMCloud.ServiceEndpoints {
+			switch endpointConfig.Name {
+			case "cis":
+				serviceEndpointOverrides.CIS = endpointConfig.URL
+			case "dnsservices":
+				serviceEndpointOverrides.DNS = endpointConfig.URL
+			case "iam":
+				serviceEndpointOverrides.IAM = endpointConfig.URL
```

Comment:
Are there named constants for "cis", "dnsservices", or "iam" that we could use?  For example, for AWS, we use named constants from aws-sdk-go.  If the IBM Cloud SDK doesn't define named constants for these service names, maybe we should define them in openshift/api.

Reply:
While I may not have a complete list of these required service names (`cis`, `iam`, etc.) yet, as we work to close on all of the Operator changes necessary for these service endpoint overrides, I have opened a PR to get these service names defined, if they can be merged and pulled in in time for FF.
https://github.com/openshift/library-go/pull/1612